### ISSUE TITLE
Allow pathlib.Path objects in surface.load_surf_mesh

### ISSUE THREAD

**Issue by alexisthual** [nilearn/nilearn#3373]
At the moment, giving `surface.load_surf_mesh()` a `pathlib.Path` object in will yield the following error:
> ```ValueError: The input type is not recognized. Valid inputs are one of the following file formats: .gii, .gii.gz, Freesurfer specific files such as .orig, .pial, .sphere, .white, .inflated or two Numpy arrays organized in a list, tuple or a namedtuple with the fields "coordinates" and "faces"```

I think it's a misleading error message and that we should allow `Path`s to be used as inputs.

**Comment by michellewang** [comment]
Hi, it looks like this has already been implemented (in PRs #3300, #3327). With `nilearn 0.10.0` I'm able to use `pathlib.Path` objects to get a `Mesh` object:

```python
from pathlib import Path
from nilearn import datasets, surface

fsaverage = datasets.fetch_surf_fsaverage()
path_pial_left = Path(fsaverage['pial_left'])
surface.load_surf_mesh(path_pial_left)
```

Maybe this issue can be closed?

**Comment by jeromedockes** [comment]
ah great, I hadn't realized that -- thanks @michellewang !
